velox/common/caching/GroupTracker.h

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/common/base/Bloom.h"
+#include "velox/common/caching/FileIds.h"
+#include "velox/common/caching/ScanTracker.h"
+#include "velox/common/caching/StringIdMap.h"
+
+#include <folly/container/F14Map.h>
+#include <folly/container/F14Set.h>
+
+namespace facebook::velox::cache {
+
+class ApproxCounter {
+ public:
+  ApproxCounter(int32_t range) : bits_(std::max(2, range / 8)) {}
+
+  void add(uint64_t value) {
+    auto hash = folly::hasher<uint64_t>()(value);
+    if (!Bloom::test(bits_.data(), bits_.size(), hash)) {
+      Bloom::set(bits_.data(), bits_.size(), hash);
+      ++count_;
+    }
+  }
+
+  int32_t count() const {
+    return count_;
+  }
+
+ private:
+  std::vector<uint64_t> bits_;
```

Comment:
- Why is this using raw bits_ vector instead of an instance of the BloomFilter class?
- Memory used by bloom filters is not accounted for. Is this intentional? Why?

Reply:
The BF memory is marginal. Everything else is 100x bigger. Use Bloom class.